ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/ScanConfigs/Wizard/ScanConfigWizardForm.tsx

```diff
@@ -73,24 +74,51 @@ function ScanConfigWizardForm({ initialFormValues }: ScanConfigWizardFormProps):
         }
     }
 
+    function wizardStepChanged() {
+        handleProfilesUpdate();
+        setCreateScanConfigError('');
+    }
+
+    function scrollToAlert() {
+        if (alertRef.current) {
+            alertRef.current.scrollIntoView({
+                behavior: 'smooth',
+                block: 'start',
+            });
+        }
+    }
+
     function onClose(): void {
         history.push(complianceEnhancedScanConfigsPath);
     }
 
     function setAllFieldsTouched(formikGroupKey: string): void {
-        const fields = Object.keys(formik.values[formikGroupKey]);
-        const touchedState = fields.reduce((acc, field) => ({ ...acc, [field]: true }), {});
-        formik.setTouched({ [formikGroupKey]: touchedState });
+        const groupHasNestedFields =
+            typeof formik.values[formikGroupKey] === 'object' &&
+            !Array.isArray(formik.values[formikGroupKey]);
+        let touchedState;
+
+        if (groupHasNestedFields) {
+            touchedState = Object.keys(formik.values[formikGroupKey]).reduce((acc, field) => {
+                acc[field] = true;
+                return acc;
+            }, {});
+            formik.setTouched({ ...formik.touched, [formikGroupKey]: touchedState });
+        } else {
+            formik.setTouched({ ...formik.touched, [formikGroupKey]: true });
+        }
     }
 
     function proceedToNextStepIfValid(
         navigateToNextStep: () => void,
         formikGroupKey: string
     ): void {
-        if (Object.keys(formik.errors?.[formikGroupKey] || {}).length === 0) {
+        const hasNoErrors = Object.keys(formik.errors?.[formikGroupKey] || {}).length === 0;
+        if (hasNoErrors) {
             navigateToNextStep();
         } else {
             setAllFieldsTouched(formikGroupKey);
+            scrollToAlert();
```

Comment:
If I'm following the code correctly, it looks like this always gets called, and then this check in the function
```
if (alertRef.current) {
```

causes this to be a no-op for the Parameters step.

During a future tech-debt paydown period, maybe we should generalize this in all our forms to scroll up to errors.

Reply:
Yeah you're following correctly. It does seem like it might be a good thing to include in forms by default, especially if we start using this pattern everywhere – the pattern where next/submit is not disabled and instead it exposes validation errors.